### PR TITLE
(Feature/GuidedTour) Unskip at register

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

If we deploy the guided tour as it is on feature/guided-tour branch, every super admin will see the guided tour
We want to show guided tour only for new super admin

## Solution

isSkipped is starting true
setSkipped to false during registration if user is super admin

### CODE REVIEW

File: packages/core/admin/admin/src/components/GuidedTour/init.js [hunk 1]
```diff
@@ -26,8 +26,8 @@ const init = initialState => {
     persistStateToLocaleStorage.addCurrentStep(null);
   }
 
-  if (skippedLocaleStorage) {
-    set(copyInitialState, 'isSkipped', true);
+  if (skippedLocaleStorage !== null) {
+    set(copyInitialState, 'isSkipped', skippedLocaleStorage);
```
Comment (soupette) [inline packages/core/admin/admin/src/components/GuidedTour/init.js:30]: Don't you need to parse the value?

File: packages/core/admin/admin/src/pages/AuthPage/index.js [hunk 2]
```diff
@@ -176,6 +178,18 @@ const AuthPage = ({ hasAdmin, setHasAdmin }) => {
...
       setHasAdmin(true);
+
+      const { roles } = user;
+
+      if (roles) {
+        const isUserSuperAdmin = roles.find(({ code }) => code === 'strapi-super-admin');
+
+        if (isUserSuperAdmin) {
+          persistStateToLocaleStorage.setSkipped(false);
+          setSkipped(false);
```
Comment (soupette) [inline packages/core/admin/admin/src/pages/AuthPage/index.js:189]: I am not sure to understand why you need to set is to false here, can't you just set the initialState to false?

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/core/admin/admin/src/components/GuidedTour/init.js [hunk 3]
```diff
@@ -26,8 +26,8 @@ const init = initialState => {
     persistStateToLocaleStorage.addCurrentStep(null);
   }
 
-  if (skippedLocaleStorage) {
-    set(copyInitialState, 'isSkipped', true);
+  if (skippedLocaleStorage !== null) {
```
[the remark this answers is not in the document]
Comment (ronronscelestes) [reply packages/core/admin/admin/src/components/GuidedTour/init.js:29]: true or false!
but in some occasions skippedLocaleStorage is null when it doesn't exist in the localeStorage, that's why I did this check, WDYT?

File: packages/core/admin/admin/src/components/GuidedTour/init.js (hunk 1 above)
Comment (ronronscelestes) [reply packages/core/admin/admin/src/components/GuidedTour/init.js:30]: already parsed with `const skippedLocaleStorage = persistStateToLocaleStorage.get(SKIPPED)`
```
  get: item => {
    return parse(localStorage.getItem(item));
  },
```

File: packages/core/admin/admin/src/components/GuidedTour/init.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (soupette) [reply packages/core/admin/admin/src/components/GuidedTour/init.js:29]: I think it is undefined in that case and not `null` right?

[the remark this answers is not in the document]
Comment (ronronscelestes) [reply packages/core/admin/admin/src/components/GuidedTour/init.js:29]: actually I was surprised but I receive null (you can log it when you arrive first time on the register page)

**Review by vincentbpro — APPROVED** [review]

**Pull request merged**